tests/integration/ha_tests/helpers.py

```diff
@@ -274,25 +279,18 @@ async def mongod_ready(ops_test: OpsTest, unit: int) -> bool:
 
 async def get_replica_set_primary(ops_test: OpsTest) -> str:
     """Returns the primary unit name based no the replica set host."""
-    rs_status = await run_mongo_op(ops_test, "rs.status()")
-    assert rs_status.succeeded, "mongod had no response for 'rs.status()'"
+    with await get_mongo_client(ops_test) as client:
+        data = client.admin.command("replSetGetStatus")
```

Comment:
`run_mongo_op()` was very flakey on my machine. Is it OK to just call the units directly instead?

Reply:
oh wow this works? It used to be the case that you couldn't run this since it was coming from outside the namespace or something like that. If this works definitely do this.